data/lib/core/creature.lua

```diff
@@ -50,3 +50,38 @@ end
 function Creature.isTile(self)
 	return false
 end
+
+function Creature:setMonsterOutfit(monster, time)
+	local monsterType = MonsterType(monster)
+	if not monsterType then
+		return false
+	end
+
+	if self:getPlayer() and not (getPlayerFlagValue(self, PlayerFlag_CanIllusionAll) or monsterType:isIllusionable()) then
```

Comment:
Ye I wanted to move it but couldn't find a "clean" way to do it while keeping backwards compat, but might give it another try soon :P